### PR TITLE
Add `bytemuck` support for `Compose`, `Extend`, `Fill`, `Mix`

### PR DESCRIPTION
**Pull request by waywardmonkeys**
This is optional.

### CODE REVIEW

File: src/impl_bytemuck.rs [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+        // Don't need to compare against MIN_VALUE as this is u8 and 0 is the MIN_VALUE.
+        *bits <= Self::MAX_VALUE
+    }
+}
+
+// Safety: The enum is `repr(u8)`. All values are `u8` and fall within
+// the min and max values.
+unsafe impl bytemuck::Contiguous for Compose {
+    type Int = u8;
+    const MIN_VALUE: u8 = Self::Clear as u8;
```
Comment (DJMcNab) [inline src/impl_bytemuck.rs:29]: I wonder if we should validate that this is actually zero somewhere? I.e. `assert!(Compose::MIN_VALUE==0)`
(or a const-permitted equivalent) in the corresponding `const _: ()`

**Review by DJMcNab — APPROVED** [review]
As an optional feature, this is fine to have

File: src/impl_bytemuck.rs (hunk 1 above)
Comment (waywardmonkeys) [reply src/impl_bytemuck.rs:29]: Handled essentially by checking that `zeroed` is `Self::Clear` below ... and this is u8, so won't be anything lower.

**Comment by waywardmonkeys** [comment]
I'm pretty sure we'll end up wanting to use this from Vello at some point once we start cleaning up some of that code. Vello is already using bytemuck.

File: src/impl_bytemuck.rs (hunk 1 above)
Comment (DJMcNab) [reply src/impl_bytemuck.rs:29]: Grand. That works for me. Thanks

**Pull request merged**